### PR TITLE
feat: Notebooks History

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We autosave changes which is great but sometimes you might make a change and think "whoops - I want to revert". 
Undo redo works whilst the Notebook is open but after that you might want to revert to a previous version.

## Changes

* Adds a popout to the sidebar for the "history" using the Activity API.
* Selecting an option previews it, which copies the old content out and saves forward.
* Fixes scrolling to a node, to only do it when the settings are opened.

## TODO
- [x] Actually implement the revert
- [x] Fix the UI / styling
- [x] Maybe preview the change? 

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/notebooks/Notebook/NotebookHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        setLocalContent(previewContent)
+        clearPreviewContent()
+        setShowHistory(false)
+    }
+
+    return (
+        <LemonBanner type="info" className="my-4">
+            <span className="flex items-center gap-2">
+                <span className="flex-1">
+                    <b>Hello time traveller!</b>
```
Comment (pauldambra) [inline frontend/src/scenes/notebooks/Notebook/NotebookHistory.tsx:118]: Need a maxty mcfly 🤣

File: frontend/src/scenes/notebooks/Notebook/notebookHistoryLogic.tsx
Comment (pauldambra) [inline frontend/src/scenes/notebooks/Notebook/notebookHistoryLogic.tsx:1]: 🚫

**Review by pauldambra — APPROVED** [review]
This is awesome... a few nitpicks but otherwise 🚢 

If there's only a single version and I open the history, then I can click on it and am offered the opportunity to revert. _Really_ the current version shouldn't offer that option

----

My first instinct to close the history was to repeat the action I'd done to open it...



---

For 100% score on the PR it'd be awesome if the activity log distinguished between a revert and a not-revert

Screenshot 2023-09-28 at 10 18 06

But that's very finicky of me

File: frontend/src/scenes/notebooks/Notebook/NotebookHistory.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Notebook/NotebookHistory.tsx:118]: :D

**Comment by benjackwhite** [comment]
> If there's only a single version and I open the history, then I can click on it and am offered the opportunity to revert. _Really_ the current version shouldn't offer that option

That's a good point! Modified it to show the "created this" state for the very first activity and mark the "current" version as current so that you can't revert to it.


> My first instinct to close the history was to repeat the action I'd done to open it...

It is now a toggle

> For 100% score on the PR it'd be awesome if the activity log distinguished between a revert and a not-revert

Yeah I thought about this. The tricky bit is it is more of a "revert commit" so there's no current way of distinguishing it.
We could perhaps add some meta fields to the `content` but I want to avoid making a schema decision here. Good idea for follow up though.

If we had a field to say something like "version this was copied from" that could be stored somewhere, at least in the activity log maybe 🤔

**Comment by pauldambra** [comment]
Ah, I was too slow :D

A couple of nits, but not blocking so 🤷 

* when you create a notebook from a previous version the toast says you created it from a template, but you didn't
* if you have the history panel open and edit the notebook, it doesn't automatically reload, so you have to close and re-open it to see the current history

**Pull request merged**